implementations/rust/ockam/ockam_examples/example_projects/tunnel/src/bin/server.rs

```diff
@@ -0,0 +1,83 @@
+use nanorand::{tls_rng, Rng};
+use ockam::{
+    Context, Entity, Result, Routed, SecureChannels, TcpTransport, TrustEveryonePolicy, Vault,
+    Worker,
+};
+
+struct ConnectionBrokerWorker {
+    tcp: TcpTransport,
+    ports: Vec<u16>,
+}
+
+impl ConnectionBrokerWorker {
+    pub fn new(tcp: TcpTransport) -> Self {
+        Self {
+            tcp,
+            ports: (4000..4100).collect(),
+        }
+    }
+}
+
+#[ockam::worker]
+impl Worker for ConnectionBrokerWorker {
+    type Message = String;
+    type Context = Context;
+
+    async fn handle_message(
+        &mut self,
+        ctx: &mut Self::Context,
+        msg: Routed<Self::Message>,
+    ) -> Result<()> {
+        let return_route = msg.return_route();
+
+        // the incoming traffic from the inlet will be redirected
+        // to this address
+        let inlet_dst = {
+            let mut route = return_route.clone();
+            route.modify().pop_back().append(msg.as_body().as_str());
+            route
+        };
+
+        // find a free port
+        let port = {
+            if self.ports.is_empty() {
+                println!("Ran out of free ports!");
+                return Ok(());
+            }
+
+            let idx = tls_rng().generate_range(0..self.ports.len());
```

Comment:
Genuine question: is there a benefit (security?) to using a random port each time rather than just popping one off?

Reply:
Good question. My initial thought was that incrementing identifiers are usually a security risk. However, I would imagine that this wouldn't help for ports, because you can easily scan the whole region of ports and look for open ones.